applications/mne_analyze/plugins/averaging/averaging.cpp

```diff
@@ -0,0 +1,633 @@
+
+QString Averaging::getName() const
+{
+    return "Averaging";
+}
+
+//=============================================================================================================
+
+QMenu *Averaging::getMenu()
+{
+    return Q_NULLPTR;
+}
+
+//=============================================================================================================
+
+QWidget *Averaging::getView()
+{
+    m_pButterflyView = new DISPLIB::ButterflyView();
+    m_pAverageLayoutView = new DISPLIB::AverageLayoutView();
+
+    QTabWidget* pTabView = new QTabWidget();
+    QWidget* testWidget = new QWidget();
+    QVBoxLayout* testLayout = new QVBoxLayout();
+
+    pTabView->addTab(m_pButterflyView, "Butterfly View");
+    pTabView->addTab(m_pAverageLayoutView, "2D Layout View");
+
+    testLayout->addWidget(pTabView);
+    testWidget->setLayout(testLayout);
+
+    testWidget->setMinimumSize(256, 256);
+    testWidget->setFocusPolicy(Qt::TabFocus);
+    testWidget->setAttribute(Qt::WA_DeleteOnClose, false);
+
+    return testWidget;
+}
+
+//=============================================================================================================
+
+QDockWidget* Averaging::getControl()
+{
+    QDockWidget* pControl = new QDockWidget(getName());
+    QWidget* pWidget = new QWidget();
+
+    m_pLayout = new QVBoxLayout;
+    m_pTabView = new QTabWidget();
+
+    //Average Settings View
+    m_pAveragingSettingsView = new DISPLIB::AveragingSettingsView(QString("MNEANALYZE/%1").arg(this->getName()));
+
+    connect(m_pAveragingSettingsView, &DISPLIB::AveragingSettingsView::changeNumAverages,
+            this, &Averaging::onChangeNumAverages, Qt::UniqueConnection);
+    connect(m_pAveragingSettingsView, &DISPLIB::AveragingSettingsView::changeBaselineFrom,
+            this, &Averaging::onChangeBaselineFrom, Qt::UniqueConnection);
+    connect(m_pAveragingSettingsView, &DISPLIB::AveragingSettingsView::changeBaselineTo,
+            this, &Averaging::onChangeBaselineTo, Qt::UniqueConnection);
+    connect(m_pAveragingSettingsView, &DISPLIB::AveragingSettingsView::changePostStim,
+            this, &Averaging::onChangePostStim, Qt::UniqueConnection);
+    connect(m_pAveragingSettingsView, &DISPLIB::AveragingSettingsView::changePreStim,
+            this, &Averaging::onChangePreStim, Qt::UniqueConnection);
+    connect(m_pAveragingSettingsView, &DISPLIB::AveragingSettingsView::changeBaselineActive,
+            this, &Averaging::onChangeBaselineActive, Qt::UniqueConnection);
+    connect(m_pAveragingSettingsView, &DISPLIB::AveragingSettingsView::resetAverage,
+            this, &Averaging::onResetAverage, Qt::UniqueConnection);
+//    connect(m_pAveragingSettingsView, &DISPLIB::AveragingSettingsView::changeStimChannel,
+//            this, &Averaging::onChangeStimChannel);
+
+    m_pAveragingSettingsView->setProcessingMode(DISPLIB::AbstractView::ProcessingMode::Offline);
+
+    // Buttons
+    QPushButton* pButton = new QPushButton();
```

Comment:
how about we add the compute, annotation, rejection and stimcheck buttons to the AveragingSettingsView's offline mode?